### PR TITLE
draft review with queries marked for JN

### PR DESCRIPTION
**Pull request by jon8787**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
 
 ## Usage for DSPs
 
-See com.uid2.client.test.IntegrationExamples for example usage for DSPs.
+For example usage for DSPs, see com.uid2.client.test.IntegrationExamples.**(gwh/JN_01 where is this? Can we link?)**
```
Comment (jon8787) [inline README.md:38]: Yes it's https://github.com/IABTechLab/uid2-client-java/blob/master/src/test/java/com/uid2/client/test/IntegrationExamples.java

File: README.md [hunk 2]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
...
-An example application which demonstrates both Basic and Advanced usage is at [uid2-examples](https://github.com/UnifiedID2/uid2-examples/tree/main/publisher/uid2-java-test-site).
+If you're using the SDK's HTTP implementation, follow these steps.
 
-### Basic usage
 1. Create an instance of PublisherUid2Client as an instance variable.
  
    `private final PublisherUid2Client publisherUid2Client = new PublisherUid2Client(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY);`
 
-2. When a user authenticates and authorizes the creation of a UID2:
+2. When a user authenticates and authorizes the creation of a UID2: **(gwh/JN_02 action? Does the user have to do something here?)**
```
Comment (jon8787) [inline README.md:56]: This is step 1-b and 1-c here: https://github.com/IABTechLab/uid2docs/blob/main/api/v2/guides/custom-publisher-integration.md

File: README.md [hunk 3]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
...
  
    `IdentityTokens identity = publisherUid2Client.generateToken(TokenGenerateInput.fromEmail(emailAddress));`
  
-
 #### Standard Integration
-If you're using [standard (client and server) integration](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/guides/publisher-client-side.md):
+
+If you're using [standard (client and server) integration](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/guides/publisher-client-side.md), follow these steps. **(gwh/JN_03 definitely only one step?)**
```
Comment (jon8787) [inline README.md:62]: This is really step 3 after the "Basic Usage" steps above. But yes, all remaining work after this step is done by the client-side SDK.

File: README.md [hunk 4]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
...
 #### Server-Only Integration
+
 If you're using [server-only integration](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/guides/custom-publisher-integration.md):
 
 1. Store this identity as a JSON string in the user's session, using: `identity.getJsonString()`
 2. To use the user's UID2 token, use `identity.getAdvertisingToken()`
 
 3. When the user accesses another page, or on a timer, determine whether a refresh is needed:
-   1. Retrieve the identity JSON string from the user's session, then call:
+   1. Retrieve the identity JSON string from the user's session, then call: **(gwh/JN_04 should this be two steps?)**
```
Comment (jon8787) [inline README.md:74]: sorry, what do you mean here?

File: README.md [hunk 5]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
...
-   2. Determine if the identity is refreshable (ie, the refresh token hasn't expired):
+   2. Determine if the identity is refreshable (that is, the refresh token hasn't expired):
 
-      ` if (identity == null || !identity.isRefreshable()) { we must no longer use this identity (eg, remove this identity from the user's session) }`
+      ` if (identity == null || !identity.isRefreshable()) { we must no longer use this identity (for example, remove this identity from the user's session) }`
    3. Determine if a refresh is needed:
 
       `if (identity.isDueForRefresh()) {..}`
-4. If a refresh is needed, call:
+4. If a refresh is needed, call: **(gwh/JN_05 can we state what this is doing? Initiate token refresh?)**
```
Comment (jon8787) [inline README.md:83]: This is step 3 here https://github.com/IABTechLab/uid2docs/blob/main/api/v2/guides/custom-publisher-integration.md

File: README.md [hunk 6]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
...
 
-### Advanced usage
+### Advanced Usage
 
-1. Create an instance of PublisherUid2Helper as an instance variable. 
+1. Create an instance of PublisherUid2Helper as an instance variable:
 
     `private final PublisherUid2Helper publisherUid2Helper = new PublisherUid2Helper(UID2_SECRET_KEY);`
-2. When a user authenticates and authorizes the creation of a UID2:
+2. When a user authenticates and authorizes the creation of a UID2: **(gwh/JN_06 can we state what this is doing?)**
```
Comment (jon8787) [inline README.md:95]: This is step 1-b and 1-c here: https://github.com/IABTechLab/uid2docs/blob/main/api/v2/guides/custom-publisher-integration.md

File: README.md [hunk 7]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
...
 
     `EnvelopeV2 envelope = publisherUid2Helper.createEnvelopeForTokenGenerateRequest(TokenGenerateInput.fromEmail(emailAddress));`
-3. Using an HTTP client library of your choice, post this envelope to the [/v2/token/generate](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md) endpoint, with:
-   1. Headers (depending on your HTTP library, this might look something like):  
+3. Using an HTTP client library of your choice, post this envelope to the [/v2/token/generate](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md) endpoint, including headers and body: **(gwh/JN_07 just noting that URLs to endpoints doc will need to be updated when the new site goes live.)**
```
Comment (jon8787) [inline README.md:98]: can we use redirects? there will be links all over the place

File: README.md [hunk 8]
```diff
@@ -25,77 +35,84 @@ Add this dependency to your project's POM:
...
+   1. Headers: Depending on your HTTP library, this might look something like the following:  
     
       `.putHeader("Authorization", "Bearer " + UID2_API_KEY)`  
       `.putHeader("X-UID2-Client-Version", PublisherUid2Helper.getVersionHeader())`
    2. Body: `envelope.getEnvelope()`
-4. If the HTTP response status code is _not_ 200, see [Response Status Codes](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md#response-status-codes) to determine next steps. Otherwise:
+4. If the HTTP response status code is _not_ 200, see [Response Status Codes](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md#response-status-codes) to determine next steps. Otherwise: **(gwh/JN_08 can we state what the below line of code is doing?)**
 
    `IdentityTokens identity = publisherUid2Helper.createIdentityfromTokenGenerateResponse({response body}, envelope);`
```
Comment (jon8787) [inline README.md:106]: it's converting the response body from a call to /token/generate into an IdentityTokens object. That object contains the UID token and expiry times.

File: README.md [hunk 9]
```diff
@@ -123,12 +140,12 @@ If you're using [server-only integration](https://github.com/UnifiedID2/uid2docs
 
 ## Usage for UID Sharers
 
-A UID Sharer is a participant that wants to share UID2s or EUIDs with another participant. Raw UIDs must be encrypted into UID tokens before sending them to another participant. See com.uid2.client.test.IntegrationExamples (runSharingExample method) for an example usage.
+A UID2 Sharer is a participant that wants to share UID2s or EUIDs with another participant. Raw UIDs must be encrypted into UID tokens before sending them to another participant. See com.uid2.client.test.IntegrationExamples (runSharingExample method) for an example usage. **(gwh/JN_09 can we link to this?)**
```
Comment (jon8787) [inline README.md:143]: Yes, as above it's https://github.com/IABTechLab/uid2-client-java/blob/master/src/test/java/com/uid2/client/test/IntegrationExamples.java

File: README.md (hunk 1 above)
Comment (genwhittTTD) [reply README.md:38]: Added, thx.

File: README.md (hunk 2 above)
Comment (genwhittTTD) [reply README.md:56]: @jon8787 : Thx but I wasn't clear. What I meant was, what is our reader doing in this step? A statement of what "IdentityTokens identity = publisherUid2Client.generateToken(TokenGenerateInput.fromEmail(emailAddress));" does.

File: README.md (hunk 3 above)
Comment (genwhittTTD) [reply README.md:62]: Done thx.

File: README.md (hunk 5 above)
Comment (genwhittTTD) [reply README.md:83]: Reworded. Thx.

File: README.md (hunk 6 above)
Comment (genwhittTTD) [reply README.md:95]: Trying again... let's discuss.

File: README.md (hunk 7 above)
Comment (genwhittTTD) [reply README.md:98]: Thinking about that. TBD.

File: README.md (hunk 8 above)
Comment (genwhittTTD) [reply README.md:106]: Done, thx.

File: README.md (hunk 9 above)
Comment (genwhittTTD) [reply README.md:143]: Done, thx.

**Pull request merged**